types/react/index.d.ts

```diff
@@ -334,7 +334,7 @@ declare namespace React {
         displayName?: string;
     }
     function createContext<T>(
-        defaultValue: T,
+        defaultValue?: T,
```

Comment:
Wouldn't this mean I can create it without a value and every consumer gets `T` while it actually is `T | undefined`?

I think overloading it with `function createContext<T>(): Context<T | undefined>` would be the sound solution. But I guess the complaint is then that you have to check for it every time when consuming?

I'd suggest using `createContext<T>(undefined as any)` which at least makes it obvious that you're using the type-system in an unsound way. Moving this "unsoundness"? into the library definition  is a recipe for confusion.

Reply:
@davidkpiano I would appreciate if you could clear up what part of my statement was confusing to you. We had this PR a second time now and I would like to know how I could explain the current approach better.

Reply:
@eps1lon I'll answer below.